### DISCUSSION TITLE
Sharing Refseq Virus taxonomy/metadata file?

### DISCUSSION THREAD

**Discussion by haleyhallowell** [bluenote-1577/sylph#21] [Q&A]
Hello! I have been really enjoying sylph so far, thank you for a great contribution to the field! 

In the paper, i saw that you used the Refseq viral database. I am interested in using this as well, and was wondering if you'd be able to share your taxonomy/metadata file for the mpa format processing script? I have the refseq viral database built for sylph, but thought i would ask before i went through the process of building the metadata file! Thanks!

**Comment by bluenote-1577** [comment]
Hi @haleyhallowell 

I actually never built the  Refseq viral taxonomy metadata file. My results for the paper were all in the default tsv output (since I just looked at the total abundance of viruses, not specific taxa). 

If you have a viral metadata file/database, you're welcome to share it as well. 

Be careful with how viruses are processed though, in https://github.com/bluenote-1577/sylph-utils/blob/main/sylph_to_taxprof.py there is some pre-processing of viral genome names to ensure compatibility with IMG/VR. You may have to modify the script slightly to make it compatible with your format. Letm eknow if you have questions

**Reply by haleyhallowell** [reply]
Thanks so much for the reply! if i can successfully build the metadata file ill be happy to share! I took a look at the suggestion you had for building the metadata from refseq and it looks a little above my skill level. I'm thinking of parsing the species/genome level names out of the fasta headers of the .fna file. do you think something like this in a .tsv file would work? The reason i am interested in using the utility script is the ability to reformat the data and then combine multiple files together (similar to merge_metaphlan_tables.py) so the actual full taxonomy does not matter _that_ much to me 

Example header: 
>NC_001798.2_Human_herpesvirus_2_strain_HG52,_complete_genome

**Reply by bluenote-1577** [reply]
If you do not care that much about the actual taxonomy, then you can just analyze sylph's output tsv file -- look at the last column "`Contig_name` which should be `NC_001798...`. 

You can run `sylph profile database.syldb reads1.fq.sylsp reads2.fq.sylsp .... -o all_results.tsv`. So `all_results.tsv` has all profiles mixed together. 

Then you can use R/python to manipulate `all_results.tsv` by analyzing the `Sample_file` and `Contig_name` columns. If you want like a matrix of results with TAXA x SAMPLE then you can use python/R to manipulate the all_results.tsv file. Ask ChatGPT , for example :)
